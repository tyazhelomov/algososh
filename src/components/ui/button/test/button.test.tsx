import renderer from 'react-test-renderer';

import { Button } from '../button';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

describe('Тест элемента Button:', () => {
  it('Кнопка с текстом рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button text={'Кнопка'} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 

  it('Кнопка без текста рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 

  it('Заблокированная кнопка рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button text={'Кнопка'} disabled={true} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 

  it('Кнопка с индикацией загрузки рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button text={'Кнопка'} isLoader={true} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 

  it('Вызов колбека при клике на кнопку выполняется без ошибок', () => {
    window.alert = jest.fn();

    render(<Button text={'Кнопка'} onClick={() => alert('Тест')} />)

    const link = screen.getByText('Кнопка');

    fireEvent.click(link);

    expect(window.alert).toHaveBeenCalledWith('Тест');
  });
});
