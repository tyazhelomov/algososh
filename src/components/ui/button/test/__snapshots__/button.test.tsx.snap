// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест элемента Button: Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Тест элемента Button: Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест элемента Button: Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тест элемента Button: Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;
